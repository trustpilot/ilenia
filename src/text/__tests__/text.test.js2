import React from 'react';
import renderer from 'react-test-renderer';
import { LocalizationProvider, Text } from '../../';

const translations = {
  'test1': 'Just a random string',
  'test2': 'A string but with a {value} in it',
  'test3': 'A string that is sort of <b>bold</b>'
};

const setupText = (string, interpolations) => (<LocalizationProvider locale="en-US" translations={translations}>
  <Text id={string} interpolations={interpolations}/>
</LocalizationProvider>);

test('Renders a plain string', () => {
  const component = renderer.create(setupText('test1', []));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with an interpolations', () => {
  const component = renderer.create(setupText('test2', { 'value': 'huge value' }));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with HTML in it', () => {
  const component = renderer.create(setupText('test3'));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
