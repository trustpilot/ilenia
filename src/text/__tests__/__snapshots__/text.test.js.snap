// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a plain string 1`] = `"Just a random string"`;

exports[`Renders a string which uses a different placeholder syntax 1`] = `"A string that uses another syntax"`;

exports[`Renders a string with an interpolations 1`] = `"A string but with a huge value in it"`;
